datasets/de/abgeordnetenwatch/crawler.py

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+import json
+import math
+from zavod import Context
+from zavod import helpers as h
+
+
+def crawl_politician(context: Context, politician_api_url: str):
+    api_response = context.fetch_json(politician_api_url, cache_days=30)
+    politician = api_response.pop("data")
+    return politician
+
+
+def crawl(context: Context):
+    # Fetch the source data URL specified in the metadata to a local path:
+    api_response = context.fetch_json(context.dataset.data.url, cache_days=30)
+    total_results = api_response.pop("meta").pop("result").pop("total")
+    num_batches = math.ceil(total_results / 100)
+
+    bi = 0
+    while bi < num_batches:
+        context.log.info(
+            "Get Politicians from range {} to {}".format(bi * 100, (bi + 1) * 100)
+        )
+
+        if bi > 0:
+            api_response = context.fetch_json(
+                context.dataset.data.url,
+                params={"range_start": bi * 100},
+                cache_days=30,
+            )
+            # We need to always recheck total number of results because the API
+            # returns a incorrect value in the first request without range_start
+            batch_total_results = api_response.pop("meta").pop("result").pop("total")
+            if batch_total_results != total_results:
+                num_batches = math.ceil(batch_total_results / 100)
+
+        for mandate in api_response.get("data"):
+            politician = mandate.pop("politician")
+
+            # Get state from parliament info
+            parliament_info = mandate.pop("parliament_period").pop("label").split(" ")
+            state = parliament_info[0]
+
+            # Create position
+            position = h.make_position(
+                context,
+                name="Member of the {} Parliament".format(state),
+                country="Germany",
+                subnational_area=state,
```

Comment:
It looks like this data includes members of the Bundestag and members of the EU parliament so we can't give all of them a subnational_area, and we can't assume parliament_info is a state.

Let's exclude the EU parliament - we already have them via the official source so duplication isn't necessary.

Let's keep the Bundestag. Only provide a value to subnational_area when the position is for a state. You can provide None for the Bundestag.

I think it's worth making a request to the parliamentary_period api_url and via that to the parliament api_url and get the `label_external_long`

Then if you drop ` Parliament` from the position name, you'll get nice standard positions like "Member of the Bundestag" or "Member of the Bürgerschaft Bremen" automatically

If you make sure those requests are cached you'll hit the cache a lot.

Reply:
I'm getting the `label_external_long` as you suggested now and using subnational_are as None when the parliament is from the Bundestag. I'm also not getting the EU parliament anymore.